back/qc_api/views/snippet/snippet_views.py

```diff
@@ -0,0 +1,53 @@
+from rest_framework.authentication import SessionAuthentication, BasicAuthentication
+from rest_framework.permissions import IsAuthenticated
+from ...models import Snippet
+from ...serializers import SnippetSerializer, SnippetScopeSerializer, SnippetBuySerializer, SnippetSellSerializer, SnippetAmountSerializer
+from ...util.decorator import catch_bad_request
+from rest_framework.decorators import api_view, permission_classes, authentication_classes
+from rest_framework.response import Response
+from rest_framework import status
+from typing import Dict, Any
+
+
+def type_extract(data) -> (str, Dict[str, Any]):
+    """get request data and extract type"""
+    try:
+        snippet_type = data.get("type")
+        return snippet_type, data
+    except KeyError:
+        raise KeyError
+
+
+def check_type_and_serialize_request(snippet_type: str, data: dict) -> Any:
```

Comment:
```suggestion
def check_type_and_serialize_request(snippet_type: str, data: Dict[Any, Any]) -> ModelSerializer:
```
Need to import ModelSerializer beforehand

Reply:
#52